bool enable
float64 frequency
---
string msg
bool ret